packages/tools/client-debug-view/src/components/ContainerSelectionDropdown.tsx

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Dropdown, IDropdownOption, IDropdownStyles, IStackTokens, Stack } from "@fluentui/react";
+import React from "react";
+
+import {
+	DebuggerRegistry,
+	IFluidClientDebugger,
+	getDebuggerRegistry,
+	getFluidClientDebuggers,
+} from "@fluid-tools/client-debugger";
+
+import { HasClientDebugger } from "../CommonProps";
+
+/**
+ * {@link ContainerSelectionDropdown} input props.
+ */
+export type ContainerSelectionDropdownProps = HasClientDebugger;
+
+/**
+ * Small header that displays core container data.
+ *
+ * @param props - See {@link ContainerSelectionDropdownProps}.
+ */
+export function ContainerSelectionDropdown(
+	props: ContainerSelectionDropdownProps,
+): React.ReactElement {
+	const dropdownStyles: Partial<IDropdownStyles> = {
+		dropdown: { width: "300px", zIndex: "1" },
+	};
+
+	const stackTokens: IStackTokens = { childrenGap: 20 };
+
+	const debuggerRegistry: DebuggerRegistry = getDebuggerRegistry();
+
+	function renewContainerOptions(debuggers: IFluidClientDebugger[]): IDropdownOption[] {
+		const options: IDropdownOption[] = [];
+		for (const x of debuggers) {
+			console.log(x.containerId);
+			options.push({
+				key: x.containerId,
+				text: x.containerNickname ?? x.containerId,
+			});
+		}
+		return options;
+	}
+
+	const [clientDebuggers, setClientDebuggers] = React.useState<IFluidClientDebugger[]>(
+		getFluidClientDebuggers(),
+	);
+
+	const [clientDebuggerOptions, setClientDebuggerOptions] = React.useState<IDropdownOption[]>(
```

Comment:
We shouldn't need to store these options as component state. Since they're strictly derived from the value of `clientDebuggers` above, we can just recompute this list each time `clientDebuggers` changes.

Reply:
Modified clientDebuggerOptions to be updated alone with clientDebuggers.